src/rpc/atmosphere.rs

```diff
@@ -31,3 +40,22 @@ impl AtmosphereService for MissionRpc {
         Ok(Response::new(res))
     }
 }
+
+fn wind2d(v: &common::Vector) -> (f32, f32) {
+    let mut heading = v.x.atan2(v.z).to_degrees();
+    if heading < 0.0 {
+        heading += 360.0;
+    }
+
+    // convert TO direction to FROM direction
+    if heading > 180.0 {
+        heading -= 180.0;
+    } else {
+        heading += 180.0;
+    }
+
+    // calc 2D strength
+    let strength = (v.z.powi(2) + v.x.powi(2)).sqrt();
+
+    (heading as f32, strength as f32)
+}
```

Comment:
Wdyt about offloading calculations like that to Rust?

Reply:
Pros:
* It is more efficient I imagine
* Moves stuff out of the MSE
Cons:
* You now need to know Rust instead of Lua

I think it is a great idea for code we are sure will not need much modification (Especially if that modificaiton is likely to be done by someone who is more familiar with Lua). I think this is a good example.